### PR TITLE
Add resume daemon for Beaker experiments.

### PR DESCRIPTION
**Pull request by brendan-ai2**
- Compatible with basic experiments created with `scripts/ai2_internal/run_with_beaker.py`.
- Intended to be run locally, e.g. via cron, to ensure experiments are resumed.

### CODE REVIEW

**Comment by brendan-ai2** [comment]
Hi @viveklak, I've written a simple tool to let AllenNLP users automatically resume preempted experiments created with our `run_with_beaker.py` script? Would you mind reviewing? I'd like to make sure I'm not doing anything gratuitously wrong with how I use the command line tools. Thanks!

File: scripts/ai2_internal/resume_daemon.py [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+logger.setLevel(logging.DEBUG)
+handler = RotatingFileHandler(
+        f"{os.environ['HOME']}/.allennlp/resume.log",
+        maxBytes=1024*1024,
+        backupCount=10)
+logger.addHandler(handler)
+
+BEAKER_QUERY_INTERVAL_SECONDS = 1
```
Comment (viveklak) [inline scripts/ai2_internal/resume_daemon.py]: Unused?

File: scripts/ai2_internal/resume_daemon.py [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+        #        "archived": false
+        #    }
+        #]
+
+        command = ["beaker", "experiment", "inspect", experiment_id]
+        experiment_json = subprocess.check_output(command)
+        experiment_data = json.loads(experiment_json)
+        # TODO(brendanr): Are these constraints ever actually violated?
+        assert len(experiment_data) == 1
+        assert len(experiment_data[0]["nodes"]) == 1
```
[the remark this answers is not in the document]
Comment (brendan-ai2) [reply scripts/ai2_internal/resume_daemon.py]: Thanks, that makes sense. I've updated the comment appropriately and documented above that this script is intended to be used with run_with_beaker.py specifically.

File: scripts/ai2_internal/resume_daemon.py (hunk 1 above)
Comment (brendan-ai2) [reply scripts/ai2_internal/resume_daemon.py]: Thanks for catching this. I've added some sleeps to smooth the load to beaker. I don't we'll ever have the number of users for this to really matter, but just as a matter of good practice I've added this in.

**Comment by brendan-ai2** [comment]
Glad to hear that automatic resume will be baked in! That'll definitely be a better solution long term. I've made a few fixes per your comments and fleshed out the install feature.

**Comment by viveklak** [comment]
LGTM

**Comment by brendan-ai2** [comment]
Thanks for the review!

**Pull request merged**